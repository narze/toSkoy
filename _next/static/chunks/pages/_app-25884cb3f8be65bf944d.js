_N_E=(window.webpackJsonp_N_E=window.webpackJsonp_N_E||[]).push([[5],{"/0+H":function(e,t,n){"use strict";t.__esModule=!0,t.isInAmpMode=u,t.useAmp=function(){return u(a.default.useContext(o.AmpStateContext))};var r,a=(r=n("q1tI"))&&r.__esModule?r:{default:r},o=n("lwAK");function u(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:{},t=e.ampFirst,n=void 0!==t&&t,r=e.hybrid,a=void 0!==r&&r,o=e.hasQuery,u=void 0!==o&&o;return n||a&&u}},0:function(e,t,n){n("74v/"),e.exports=n("nOHt")},"74v/":function(e,t,n){(window.__NEXT_P=window.__NEXT_P||[]).push(["/_app",function(){return n("cha2")}])},"8Kt/":function(e,t,n){"use strict";n("lSNA");t.__esModule=!0,t.defaultHead=d,t.default=void 0;var r,a=function(e){if(e&&e.__esModule)return e;if(null===e||"object"!==typeof e&&"function"!==typeof e)return{default:e};var t=s();if(t&&t.has(e))return t.get(e);var n={},r=Object.defineProperty&&Object.getOwnPropertyDescriptor;for(var a in e)if(Object.prototype.hasOwnProperty.call(e,a)){var o=r?Object.getOwnPropertyDescriptor(e,a):null;o&&(o.get||o.set)?Object.defineProperty(n,a,o):n[a]=e[a]}n.default=e,t&&t.set(e,n);return n}(n("q1tI")),o=(r=n("Xuae"))&&r.__esModule?r:{default:r},u=n("lwAK"),i=n("FYa8"),c=n("/0+H");function s(){if("function"!==typeof WeakMap)return null;var e=new WeakMap;return s=function(){return e},e}function d(){var e=arguments.length>0&&void 0!==arguments[0]&&arguments[0],t=[a.default.createElement("meta",{charSet:"utf-8"})];return e||t.push(a.default.createElement("meta",{name:"viewport",content:"width=device-width"})),t}function f(e,t){return"string"===typeof t||"number"===typeof t?e:t.type===a.default.Fragment?e.concat(a.default.Children.toArray(t.props.children).reduce((function(e,t){return"string"===typeof t||"number"===typeof t?e:e.concat(t)}),[])):e.concat(t)}var l=["name","httpEquiv","charSet","itemProp"];function p(e,t){return e.reduce((function(e,t){var n=a.default.Children.toArray(t.props.children);return e.concat(n)}),[]).reduce(f,[]).reverse().concat(d(t.inAmpMode)).filter(function(){var e=new Set,t=new Set,n=new Set,r={};return function(a){var o=!0;if(a.key&&"number"!==typeof a.key&&a.key.indexOf("$")>0){var u=a.key.slice(a.key.indexOf("$")+1);e.has(u)?o=!1:e.add(u)}switch(a.type){case"title":case"base":t.has(a.type)?o=!1:t.add(a.type);break;case"meta":for(var i=0,c=l.length;i<c;i++){var s=l[i];if(a.props.hasOwnProperty(s))if("charSet"===s)n.has(s)?o=!1:n.add(s);else{var d=a.props[s],f=r[s]||new Set;f.has(d)?o=!1:(f.add(d),r[s]=f)}}}return o}}()).reverse().map((function(e,t){var n=e.key||t;return a.default.cloneElement(e,{key:n})}))}function h(e){var t=e.children,n=(0,a.useContext)(u.AmpStateContext),r=(0,a.useContext)(i.HeadManagerContext);return a.default.createElement(o.default,{reduceComponentsToState:p,headManager:r,inAmpMode:(0,c.isInAmpMode)(n)},t)}h.rewind=function(){};var v=h;t.default=v},Bnag:function(e,t){e.exports=function(){throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.")}},EbDI:function(e,t){e.exports=function(e){if("undefined"!==typeof Symbol&&Symbol.iterator in Object(e))return Array.from(e)}},Ijbi:function(e,t,n){var r=n("WkPL");e.exports=function(e){if(Array.isArray(e))return r(e)}},RIqP:function(e,t,n){var r=n("Ijbi"),a=n("EbDI"),o=n("ZhPi"),u=n("Bnag");e.exports=function(e){return r(e)||a(e)||o(e)||u()}},Xuae:function(e,t,n){"use strict";var r=n("RIqP"),a=n("lwsE"),o=n("W8MJ"),u=(n("PJYZ"),n("7W2i")),i=n("a1gu"),c=n("Nsbk");function s(e){var t=function(){if("undefined"===typeof Reflect||!Reflect.construct)return!1;if(Reflect.construct.sham)return!1;if("function"===typeof Proxy)return!0;try{return Date.prototype.toString.call(Reflect.construct(Date,[],(function(){}))),!0}catch(e){return!1}}();return function(){var n,r=c(e);if(t){var a=c(this).constructor;n=Reflect.construct(r,arguments,a)}else n=r.apply(this,arguments);return i(this,n)}}t.__esModule=!0,t.default=void 0;var d=n("q1tI"),f=function(e){u(n,e);var t=s(n);function n(e){var o;return a(this,n),(o=t.call(this,e))._hasHeadManager=void 0,o.emitChange=function(){o._hasHeadManager&&o.props.headManager.updateHead(o.props.reduceComponentsToState(r(o.props.headManager.mountedInstances),o.props))},o._hasHeadManager=o.props.headManager&&o.props.headManager.mountedInstances,o}return o(n,[{key:"componentDidMount",value:function(){this._hasHeadManager&&this.props.headManager.mountedInstances.add(this),this.emitChange()}},{key:"componentDidUpdate",value:function(){this.emitChange()}},{key:"componentWillUnmount",value:function(){this._hasHeadManager&&this.props.headManager.mountedInstances.delete(this),this.emitChange()}},{key:"render",value:function(){return null}}]),n}(d.Component);t.default=f},c473:function(e,t,n){},cha2:function(e,t,n){"use strict";n.r(t),n.d(t,"default",(function(){return d}));var r=n("q1tI"),a=n.n(r),o=n("8Kt/"),u=n.n(o),i=(n("c473"),n("CjxU")),c=n("uDoD"),s=a.a.createElement;function d(e){var t=e.Component,n=e.pageProps;return s(a.a.Fragment,null,s(u.a,null,s("title",null,"toSkoy - \u0e41\u0e1b\u0e25\u0e07\u0e20\u0e32\u0e29\u0e32\u0e44\u0e17\u0e22\u0e40\u0e1b\u0e47\u0e19\u0e20\u0e32\u0e29\u0e32\u0e2a\u0e01\u0e4a\u0e2d\u0e22"),s("link",{rel:"icon",href:"/favicon.ico"}),s("link",{rel:"stylesheet",href:"https://cdn.jsdelivr.net/npm/hundred-days-of-code-css-ribbon@v0.1.1/100-days-of-code-ribbon.css"})),s(i.a,{theme:c.a},s(t,n)))}},lwAK:function(e,t,n){"use strict";var r;t.__esModule=!0,t.AmpStateContext=void 0;var a=((r=n("q1tI"))&&r.__esModule?r:{default:r}).default.createContext({});t.AmpStateContext=a}},[[0,0,1,2,3]]]);