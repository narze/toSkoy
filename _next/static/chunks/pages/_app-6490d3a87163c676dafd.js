_N_E=(window.webpackJsonp_N_E=window.webpackJsonp_N_E||[]).push([[5],{"/0+H":function(e,t,n){"use strict";t.__esModule=!0,t.isInAmpMode=u,t.useAmp=function(){return u(a.default.useContext(o.AmpStateContext))};var r,a=(r=n("q1tI"))&&r.__esModule?r:{default:r},o=n("lwAK");function u(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:{},t=e.ampFirst,n=void 0!==t&&t,r=e.hybrid,a=void 0!==r&&r,o=e.hasQuery,u=void 0!==o&&o;return n||a&&u}},0:function(e,t,n){n("74v/"),e.exports=n("nOHt")},"5fIB":function(e,t,n){var r=n("7eYB");e.exports=function(e){if(Array.isArray(e))return r(e)}},"74v/":function(e,t,n){(window.__NEXT_P=window.__NEXT_P||[]).push(["/_app",function(){return n("cha2")}])},"8Kt/":function(e,t,n){"use strict";n("oI91");t.__esModule=!0,t.defaultHead=f,t.default=void 0;var r,a=function(e){if(e&&e.__esModule)return e;if(null===e||"object"!==typeof e&&"function"!==typeof e)return{default:e};var t=s();if(t&&t.has(e))return t.get(e);var n={},r=Object.defineProperty&&Object.getOwnPropertyDescriptor;for(var a in e)if(Object.prototype.hasOwnProperty.call(e,a)){var o=r?Object.getOwnPropertyDescriptor(e,a):null;o&&(o.get||o.set)?Object.defineProperty(n,a,o):n[a]=e[a]}n.default=e,t&&t.set(e,n);return n}(n("q1tI")),o=(r=n("Xuae"))&&r.__esModule?r:{default:r},u=n("lwAK"),i=n("FYa8"),c=n("/0+H");function s(){if("function"!==typeof WeakMap)return null;var e=new WeakMap;return s=function(){return e},e}function f(){var e=arguments.length>0&&void 0!==arguments[0]&&arguments[0],t=[a.default.createElement("meta",{charSet:"utf-8"})];return e||t.push(a.default.createElement("meta",{name:"viewport",content:"width=device-width"})),t}function d(e,t){return"string"===typeof t||"number"===typeof t?e:t.type===a.default.Fragment?e.concat(a.default.Children.toArray(t.props.children).reduce((function(e,t){return"string"===typeof t||"number"===typeof t?e:e.concat(t)}),[])):e.concat(t)}var l=["name","httpEquiv","charSet","itemProp"];function p(e,t){return e.reduce((function(e,t){var n=a.default.Children.toArray(t.props.children);return e.concat(n)}),[]).reduce(d,[]).reverse().concat(f(t.inAmpMode)).filter(function(){var e=new Set,t=new Set,n=new Set,r={};return function(a){var o=!0;if(a.key&&"number"!==typeof a.key&&a.key.indexOf("$")>0){var u=a.key.slice(a.key.indexOf("$")+1);e.has(u)?o=!1:e.add(u)}switch(a.type){case"title":case"base":t.has(a.type)?o=!1:t.add(a.type);break;case"meta":for(var i=0,c=l.length;i<c;i++){var s=l[i];if(a.props.hasOwnProperty(s))if("charSet"===s)n.has(s)?o=!1:n.add(s);else{var f=a.props[s],d=r[s]||new Set;d.has(f)?o=!1:(d.add(f),r[s]=d)}}}return o}}()).reverse().map((function(e,t){var n=e.key||t;return a.default.cloneElement(e,{key:n})}))}function h(e){var t=e.children,n=(0,a.useContext)(u.AmpStateContext),r=(0,a.useContext)(i.HeadManagerContext);return a.default.createElement(o.default,{reduceComponentsToState:p,headManager:r,inAmpMode:(0,c.isInAmpMode)(n)},t)}h.rewind=function(){};var m=h;t.default=m},Xuae:function(e,t,n){"use strict";var r=n("mPvQ"),a=n("/GRZ"),o=n("i2R6"),u=(n("qXWd"),n("48fX")),i=n("tCBg"),c=n("T0f4");function s(e){var t=function(){if("undefined"===typeof Reflect||!Reflect.construct)return!1;if(Reflect.construct.sham)return!1;if("function"===typeof Proxy)return!0;try{return Date.prototype.toString.call(Reflect.construct(Date,[],(function(){}))),!0}catch(e){return!1}}();return function(){var n,r=c(e);if(t){var a=c(this).constructor;n=Reflect.construct(r,arguments,a)}else n=r.apply(this,arguments);return i(this,n)}}t.__esModule=!0,t.default=void 0;var f=n("q1tI"),d=function(e){u(n,e);var t=s(n);function n(e){var o;return a(this,n),(o=t.call(this,e))._hasHeadManager=void 0,o.emitChange=function(){o._hasHeadManager&&o.props.headManager.updateHead(o.props.reduceComponentsToState(r(o.props.headManager.mountedInstances),o.props))},o._hasHeadManager=o.props.headManager&&o.props.headManager.mountedInstances,o}return o(n,[{key:"componentDidMount",value:function(){this._hasHeadManager&&this.props.headManager.mountedInstances.add(this),this.emitChange()}},{key:"componentDidUpdate",value:function(){this.emitChange()}},{key:"componentWillUnmount",value:function(){this._hasHeadManager&&this.props.headManager.mountedInstances.delete(this),this.emitChange()}},{key:"render",value:function(){return null}}]),n}(f.Component);t.default=d},c473:function(e,t,n){},cha2:function(e,t,n){"use strict";n.r(t),n.d(t,"default",(function(){return l}));n("nKUr");var r=n("qKvR"),a=(n("psfu"),n("q1tI")),o=n.n(a),u=n("8Kt/"),i=n.n(u),c=(n("c473"),n("CjxU")),s=n("uDoD"),f={name:"utwuk8",styles:"@keyframes spin{to{transform:rotate(360deg);}}@keyframes ping{75%,100%{transform:scale(2);opacity:0;}}@keyframes pulse{50%{opacity:.5;}}@keyframes bounce{0%,100%{transform:translateY(-25%);animationTimingFunction:cubic-bezier(0.8,0,1,1);}50%{transform:none;animationTimingFunction:cubic-bezier(0,0,0.2,1);}}"},d=function(){return Object(r.d)(r.a,{styles:f})};function l(e){var t=e.Component,n=e.pageProps;return Object(r.d)(o.a.Fragment,null,Object(r.d)(i.a,null,Object(r.d)("title",null,"toSkoy - \u0e41\u0e1b\u0e25\u0e07\u0e20\u0e32\u0e29\u0e32\u0e44\u0e17\u0e22\u0e40\u0e1b\u0e47\u0e19\u0e20\u0e32\u0e29\u0e32\u0e2a\u0e01\u0e4a\u0e2d\u0e22"),Object(r.d)("link",{rel:"icon",href:"/favicon.ico"}),Object(r.d)("link",{rel:"stylesheet",href:"https://cdn.jsdelivr.net/npm/hundred-days-of-code-css-ribbon@v0.1.1/100-days-of-code-ribbon.css"})),Object(r.d)(d,null),Object(r.d)(c.a,{theme:s.a},Object(r.d)(t,n)))}},kG2m:function(e,t){e.exports=function(){throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.")}},lwAK:function(e,t,n){"use strict";var r;t.__esModule=!0,t.AmpStateContext=void 0;var a=((r=n("q1tI"))&&r.__esModule?r:{default:r}).default.createContext({});t.AmpStateContext=a},mPvQ:function(e,t,n){var r=n("5fIB"),a=n("rlHP"),o=n("KckH"),u=n("kG2m");e.exports=function(e){return r(e)||a(e)||o(e)||u()}},oI91:function(e,t){e.exports=function(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}},psfu:function(e,t,n){},rlHP:function(e,t){e.exports=function(e){if("undefined"!==typeof Symbol&&Symbol.iterator in Object(e))return Array.from(e)}}},[[0,0,1,3,2]]]);