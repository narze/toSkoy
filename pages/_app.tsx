import Head from 'next/head'
import '../styles/base.css'
import { ThemeProvider, theme } from '@chakra-ui/core'

export default function App({
  Component,
  pageProps,
}: {
  Component: any
  pageProps: any
}): JSX.Element {
  return (
    <>
      <Head>
        <title>
          Create Next App - with TypeScript, ESLint, Jest, TailwindCSS, Emotion
        </title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <ThemeProvider theme={theme}>
        <Component {...pageProps} />
      </ThemeProvider>
    </>
  )
}
