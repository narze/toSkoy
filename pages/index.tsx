/** @jsx jsx */
import { jsx, css } from '@emotion/core'
import tw from '@tailwindcssinjs/macro'
import { Textarea } from '@chakra-ui/core'
import Skoy from 'skoy'
import { useState } from 'react'

export const Home = (): JSX.Element => {
  const [input, setInput] = useState('สวัสดี')
  const [output, setOutput] = useState('ษวัษดลีร์')

  function handleInputChange(e) {
    const nextInput = e.target.value
    setInput(nextInput)
    setOutput(Skoy.convert(nextInput))
  }

  return (
    <main
      css={tw`flex justify-center items-center min-w-full h-screen text-center`}
    >
      <div css={[tw`flex flex-col`, css`min-width: 640px: max-width: 50%`]}>
        <h1 css={tw`text-6xl`}>toSkoy</h1>
        <Textarea
          textAlign="center"
          my={4}
          fontSize="2rem"
          data-testid="input"
          value={input}
          onKeyDown={handleInputChange}
          onChange={handleInputChange}
          placeholder="ภาษาไทย"
          tabIndex={0}
        />
        <Textarea
          textAlign="center"
          my={4}
          fontSize="2rem"
          data-testid="output"
          value={output}
          isReadOnly={true}
          placeholder="พ๊ษ๊สก๊อยย์"
        />
      </div>
    </main>
  )
}

export default Home
