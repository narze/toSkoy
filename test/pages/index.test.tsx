import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders with buttons alert', () => {
    const { getByText } = render(<Home />, {})

    getByText('Emotion CSS')
    getByText('Emotion React')
    getByText('Chakra-UI')
  })
})
