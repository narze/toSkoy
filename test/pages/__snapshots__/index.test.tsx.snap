// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  .emotion-9 {
  --text-opacity: 1;
  --bg-opacity: 1;
  position: relative;
  width: 16rem;
  min-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-width: 1px;
  border-color: transparent;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 500;
  border-radius: 0.375rem;
  color: #ffffff;
  color: rgba(255,255,255,var(--text-opacity));
  background-color: #4b5563;
  background-color: rgba(75,85,99,var(--bg-opacity));
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,-webkit-transform;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
}

.emotion-9:hover {
  --bg-opacity: 1;
  background-color: #6b7280;
  background-color: rgba(107,114,128,var(--bg-opacity));
}

.emotion-9:focus {
  --border-opacity: 1;
  outline: 0;
  border-color: #374151;
  border-color: rgba(55,65,81,var(--border-opacity));
  box-shadow: 0 0 0 3px rgba(159,166,178,0.45);
}

.emotion-9:active {
  --bg-opacity: 1;
  background-color: #374151;
  background-color: rgba(55,65,81,var(--bg-opacity));
}

.emotion-8 {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 0.75rem;
}

.emotion-7 {
  height: 1.25rem;
  width: 1.25rem;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,-webkit-transform;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  color: hotpink;
}

.group:hover .emotion-7 {
  --text-opacity: 1;
  color: #9fa6b2;
  color: rgba(159,166,178,var(--text-opacity));
}

.emotion-6 {
  margin-top: 0.75rem;
  margin-bottom: 0.75rem;
  font-size: 1.125rem;
}

.emotion-0 {
  --bg-opacity: 1;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  background-color: #16bdca;
  background-color: rgba(22,189,202,var(--bg-opacity));
  border-width: 1px;
  border-color: transparent;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 500;
  border-radius: 0.375rem;
}

.emotion-14 {
  margin-top: 2rem;
  display: grid;
  grid-gap: 1rem;
  gap: 1rem;
  grid-template-columns: repeat(1,minmax(0,1fr));
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-12 {
  --text-opacity: 1;
  --bg-opacity: 1;
  position: relative;
  width: 16rem;
  min-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-width: 1px;
  border-color: transparent;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 500;
  border-radius: 0.375rem;
  color: #ffffff;
  color: rgba(255,255,255,var(--text-opacity));
  background-color: #047481;
  background-color: rgba(4,116,129,var(--bg-opacity));
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,-webkit-transform;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
}

.emotion-12:hover {
  --bg-opacity: 1;
  background-color: #0694a2;
  background-color: rgba(6,148,162,var(--bg-opacity));
}

.emotion-12:focus {
  --border-opacity: 1;
  outline: 0;
  border-color: #036672;
  border-color: rgba(3,102,114,var(--border-opacity));
  box-shadow: 0 0 0 3px rgba(126,220,226,0.45);
}

.emotion-12:active {
  --bg-opacity: 1;
  background-color: #036672;
  background-color: rgba(3,102,114,var(--bg-opacity));
}

.emotion-11 {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 0.75rem;
}

.emotion-10 {
  height: 1.25rem;
  width: 1.25rem;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,-webkit-transform;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  color: hotpink;
}

.group:hover .emotion-10 {
  --text-opacity: 1;
  color: #16bdca;
  color: rgba(22,189,202,var(--text-opacity));
}

.emotion-13 {
  border-radius: 0.25rem;
  font-weight: 600;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  line-height: 1.2;
  outline: none;
  height: 2.5rem;
  min-width: 2.5rem;
  font-size: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  background-color: #319795;
  color: #fff;
}

.emotion-13:hover {
  background-color: #2C7A7B;
}

.emotion-13:focus {
  box-shadow: 0 0 0 3px rgba(66,153,225,0.6);
}

.emotion-13:active,
.emotion-13[data-active=true] {
  background-color: #285E61;
}

.emotion-13:disabled,
.emotion-13:disabled:focus,
.emotion-13:disabled:hover,
.emotion-13[aria-disabled=true],
.emotion-13[aria-disabled=true]:focus,
.emotion-13[aria-disabled=true]:hover {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: none;
}

<div
    class="jsx-2250570153 container"
  >
    <main
      class="jsx-2250570153"
    >
      <h1
        class="jsx-2250570153 title"
      >
        Welcome to 
        <a
          class="jsx-2250570153"
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="jsx-2250570153 emotion-6"
        data-testid="subtitle"
      >
        with
        <span
          class="jsx-2250570153 emotion-0"
        >
          TypeScript
        </span>
        <span
          class="jsx-2250570153 emotion-0"
        >
          ESLint
        </span>
        <span
          class="jsx-2250570153 emotion-0"
        >
          Jest
        </span>
        <span
          class="jsx-2250570153 emotion-0"
        >
          TailwindCSS
        </span>
        <span
          class="jsx-2250570153 emotion-0"
        >
          Emotion
        </span>
        and
        <span
          class="jsx-2250570153 emotion-0"
        >
          Cypress
        </span>
      </p>
      <p
        class="jsx-2250570153 description"
      >
        Get started by editing 
        <code
          class="jsx-2250570153"
        >
          pages/index.tsx
        </code>
      </p>
      <div
        class="jsx-2250570153 emotion-14"
      >
        <button
          class="emotion-9 group"
        >
          <span
            class="emotion-8"
          >
            <svg
              class="emotion-7"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                clip-rule="evenodd"
                d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          Emotion CSS
        </button>
        <button
          class="group  emotion-12"
        >
          <span
            class="emotion-11"
          >
            <svg
              class="emotion-10"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                clip-rule="evenodd"
                d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          Emotion React
        </button>
        <button
          class="emotion-13"
          type="button"
        >
          Chakra-UI
        </button>
      </div>
      <div
        class="jsx-2250570153 grid"
      >
        <a
          class="jsx-2250570153 card"
          href="https://nextjs.org/docs"
        >
          <h3
            class="jsx-2250570153"
          >
            Documentation →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="jsx-2250570153 card"
          href="https://nextjs.org/learn"
        >
          <h3
            class="jsx-2250570153"
          >
            Learn →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="jsx-2250570153 card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h3
            class="jsx-2250570153"
          >
            Examples →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="jsx-2250570153 card"
          href="https://vercel.com/import?filter=next.js&utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h3
            class="jsx-2250570153"
          >
            Deploy →
          </h3>
          <p
            class="jsx-2250570153"
          >
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="jsx-2250570153"
    >
      <a
        class="jsx-2250570153"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <img
          alt="Vercel Logo"
          class="jsx-2250570153 logo"
          src="/vercel.svg"
        />
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
